### DISCUSSION TITLE
How to run program without command with arguments

### DISCUSSION THREAD

**Discussion by Gigas002** [spectreconsole/spectre.console#733] [Q&A] [answered]
I've searched through docs and examples and there's no such example. What I want is to run a simple program like this:

```text
app --input "/home/user/input" --output "/home/user/output"
```

In other words, my point is only to parse the args. How can this be achieved? Also, how can I make these arguments required, but not positional? Adding attribute:

```c#
[CommandOption("-i|--input [INPUT]")]
```

throws an error: `Error: The option 'input' has an optional value but does not implement IFlagValue`.

I also can't explicitly add `ValueIsOptional` this way: 

```c#
[CommandOption("-i|--input [INPUT]", ValueIsOptional = true)]
```

the app just won't compile.

**Comment by patriksvensson** [comment, marked as answer]
> app --input "/home/user/input" --output "/home/user/output"
> In other words, my point is only to parse the args. How can this be achieved? 

See https://spectreconsole.net/api/spectre.console.cli/commandapp_1/

>Also, how can I make these arguments required, but not positional? Adding attribute:
>
>[CommandOption("-i|--input [INPUT]")]
>throws an error: Error: The option 'input' has an optional value but does not implement IFlagValue.

Use `[CommandOption("-i|--input <INPUT>")]` and do validation manually by overriding `Validate` in either the settings class or your command. Options are always optional, so any additional logic will need to be done there.

**Reply by patriksvensson** [reply]
You can also read about default commands here: https://spectreconsole.net/cli/commandapp (first section)

**Reply by Gigas002** [reply]
Thanks for quick response, I'll try out these changes

**Comment by MeikelLP** [comment]
@patriksvensson I want to write unit tests for my arguments/options:

Here is an example of what I want to achieve:

```cs

var settings = new CommandApp().ParseArguments<MySettings>("--verbose --provider Abc");

Assert.True(settings.IsVerbose);
Assert.Equal(ProviderEnum.Abc, settings.Provider);
```

How can I achieve this? I do NOT want to execute the command nor do I want to write wrapper commands for these cases.

**Reply by patriksvensson** [reply]
Not at a computer today, but take a look at our unit tests which does what you're trying to achieve.
